doc/security/sensor-threat.rst

```diff
@@ -0,0 +1,327 @@
+   contain hardware that maintains a secure time. However, most SoCs in
+   use do not have support for this, and it will be necessary to consult
+   an external time service.
+   `RFC4330 <https://tools.ietf.org/html/rfc4330>`__ and referenced RFCs
+   describe the Simple Network Time Protocol that can be used to query
+   the current time from a network time server.
+
+5. **Device lifecycle**. And IoT device will have a lifecycle from
+   production to destruction and disposal of the device. Aspects of this
+   lifecycle that impact security include initial provisioning, normal
+   operation, re-provisioning, and destruction.
+
+   a. **Initial provisioning**. During the initial provisioning stage,
+      it is necessary to program the bootloader, an initial application
+      image, a device secret, and initial configuration data. In
+      addition, the bootloader flash protection shall be installed. Of
+      this information, only the device secret needs to differ per
+      device. This secret shall be securely maintained, and destroyed in
+      all locations outside of the device once it has been programmed.
+
+   b. **Normal operation**. Normal operation includes the behavior
+      described by the rest of this document.
+
+   c. **Re-provisioning**. Sometimes it is necessary to re-provision a
+      device, such as for a different application. One way to do this is
+      to keep the same device secret, and replace the configuration
+      data, as well as the cloud service data associated with the
+      device. It is also possible to program a new device secret, but if
+      this is done it shall be done securely, and the new secret
+      destroyed externally once programmed into the device.
+
+   d. **Destruction**. To prevent the device secret from being used to
+      spoof the device, upon decommissioning, the secret for a
+      particular device shall be rendered ineffective. Possibilities
+      include:
+
+      i.    **Hardware destruction of the device**.
+
+      ii.   **Securely wiping the flash area containing the
+            secret** [3]_.
+
+      iii.  **Removing the device identity and certificate from the
+            service**.
+
+Other Considerations
+====================
+
+In addition to the above, network connected devices generally will need
+a way to configure them to connect to the network environment they are
+placed in. There are numerous ways of doing this, and it is important
+for these configuration methods to not circumvent the security
+requirements described above.
+
+.. [1]
+   See https://www.slideshare.net/kartben/iot-developer-survey-2018. As
+   of this writing, the three major cloud IoT service providers, AWS
+   IoT, Google Cloud IoT, and Microsoft Azure IoT all provide MQTT over
+   TLS. Some feedback has suggested that some find difficulty with UDP
+   protocols and routing issues on various networks.
+
+.. [2]
+   What is considered secure changes as new exploits are discovered.
```

Comment:
I had to read this sentence a few times to parse it.  How about:

    As new exploits are discovered, what is considered secure can change.